mlir/lib/Dialect/Vector/Transforms/VectorLinearize.cpp

```diff
@@ -415,6 +419,35 @@ struct LinearizeVectorBitCast final
   }
 };
 
+/// This pattern converts the SplatOp to work on a linearized vector.
+/// Following,
+///   vector.splat %value : vector<4x4xf32>
+/// is converted to:
+///   %out_1d = vector.splat %value : vector<16xf32>
+///   %out_nd = vector.shape_cast %out_1d : vector<16xf32> to vector<4x4xf32>
+/// It ensures that the operation is compatible with the target vector
```

Comment:
This info about bit width is no longer relevant since https://github.com/llvm/llvm-project/pull/136581

Reply:
oh my bad, fixed